osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -70,12 +70,8 @@ protected override void LoadComplete()
             client.LoadRequested += onLoadRequested;
             client.RoomUpdated += onRoomUpdated;
 
-            isConnected.BindTo(client.IsConnected);
-            isConnected.BindValueChanged(connected =>
-            {
-                if (!connected.NewValue)
-                    handleRoomLost();
-            }, true);
```

Comment:
As mentioned in the commit message, this isn't required as there's a second path to `handleRoomLost` on `Room` becoming null.

Leaving this in place causes the whole flow to fail because it handles the reconnect as if a disconnection happened, booting straight out of the room.

Reply:
Hmm, I'm failing to see any difference between the above callback and `MultiplayerClient`'s. If the user is in the room, they shouldn't be reconnecting unless they leave and create another room?

Reply:
Phrasing another way: If this callback is left in place, after everything succeeds (reconnect done, use joined newly created room on new server) it will *then* call `handleRoomLost` and exit out of everything.